src/helpers/schema/replace-vars-helper.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Yoast\WP\SEO\Helpers\Schema;
+
+use WPSEO_Replace_Vars;
+use Yoast\WP\SEO\Conditionals\No_Conditionals;
+use Yoast\WP\SEO\Config\Schema_IDs;
+use Yoast\WP\SEO\Memoizers\Meta_Tags_Context_Memoizer;
+use Yoast\WP\SEO\Presentations\Indexable_Presentation;
+
+/**
+ * Registers the Schema replace variables and exposes a method to replace variables on a Schema graph.
+ */
+class Replace_Vars_Helper {
+
+	use No_Conditionals;
+
+	/**
+	 * The meta tags context memoizer.
+	 *
+	 * @var Meta_Tags_Context_Memoizer
+	 */
+	protected $meta_tags_context_memoizer;
+
+	/**
+	 * The replace vars.
+	 *
+	 * @var WPSEO_Replace_Vars
+	 */
+	protected $replace_vars;
+
+	/**
+	 * The Schema ID helper.
+	 *
+	 * @var ID_Helper
+	 */
+	protected $id_helper;
+
+	/**
+	 * Replace_Vars_Helper constructor.
+	 *
+	 * @param Meta_Tags_Context_Memoizer $meta_tags_context_memoizer The meta tags context memoizer.
+	 * @param WPSEO_Replace_Vars         $replace_vars               The replace vars.
+	 * @param ID_Helper                  $id_helper                  The Schema ID helper.
+	 */
+	public function __construct(
+		Meta_Tags_Context_Memoizer $meta_tags_context_memoizer,
+		WPSEO_Replace_Vars $replace_vars,
+		ID_Helper $id_helper
+	) {
+		$this->meta_tags_context_memoizer = $meta_tags_context_memoizer;
+		$this->replace_vars               = $replace_vars;
+		$this->id_helper                  = $id_helper;
+	}
+
+	/**
+	 * Replaces the variables.
+	 *
+	 * @param array                  $schema_data  The Schema data.
+	 * @param Indexable_Presentation $presentation The indexable presentation.
+	 *
+	 * @return array The array with replaced vars.
+	 */
+	public function replace( array $schema_data, Indexable_Presentation $presentation ) {
+		foreach ( $schema_data as $key => $value ) {
+			if ( \is_array( $value ) ) {
+				$schema_data[ $key ] = $this->replace( $value, $presentation );
+
+				continue;
+			}
+
+			$schema_data[ $key ] = $this->replace_vars->replace( $value, $presentation->source );
+		}
+
+		return $schema_data;
+	}
+
+	/**
+	 * Registers the Schema-related replace vars.
+	 *
+	 * @return void
+	 */
+	public function register_replace_vars() {
```

Comment:
if you add a $context parameter, that makes this method super easy to test. Also, please add unit tests 🙃

Reply:
I ultimately decided to go the path I went before: just register the replace vars when the meta tags context is known.